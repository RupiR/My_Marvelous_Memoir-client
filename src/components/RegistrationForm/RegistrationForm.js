import React, { Component } from 'react'
import { Button, Input, Required } from '../Utils/Utils'
import AuthApiService from '../../services/auth-api-service'

export default class RegistrationForm extends Component {
  static defaultProps = {
    onRegistrationSuccess: () => { }
  }

  state = { error: null }

  handleSubmit = ev => {
    ev.preventDefault()
    const { full_name, email, username, password } = ev.target

    this.setState({ error: null })
    AuthApiService.postUser({
      username: username.value,
      password: password.value,
      full_name: full_name.value,
      email: email.value,
    })
      .then(user => {
        full_name.value = ''
        email.value = ''
        username.value = ''
        password.value = ''
        this.props.onRegistrationSuccess()
      })
      .catch(res => {
        this.setState({ error: res.error })
      })
  }

  render() {
    const { error } = this.state
    return (
      <form
        className='RegistrationForm'
        onSubmit={this.handleSubmit}
      >
        <div role='alert'>
          {error && <p className='red'>{error}</p>}
        </div>
        <div className='full_name'>
          <label htmlFor='RegistrationForm__full_name'>
            Full name <Required />
          </label>
          <Input
            name='full_name'
            type='text'
            required
            id='RegistrationForm__full_name'>
          </Input>
        </div>
        <div className='username'>
          <label htmlFor='RegistrationForm__username'>
            User name <Required />
          </label>
          <Input
            name='username'
            type='text'
            required
            id='RegistrationForm__username'>
          </Input>
        </div>
        <div className='password'>
          <label htmlFor='RegistrationForm__password'>
            Password <Required />
          </label>
          <Input
            name='password'
            type='password'
            required
            id='RegistrationForm__password'>
          </Input>
        </div>
        <div className='email'>
          <label htmlFor='RegistrationForm__email'>
            Email <Required />
          </label>
          <Input
            name='email'
            type='text'
            required
            id='RegistrationForm__email'>
          </Input>
        </div>
        <Button type='submit'>
          Register
        </Button>
      </form>
    )
  }
}
