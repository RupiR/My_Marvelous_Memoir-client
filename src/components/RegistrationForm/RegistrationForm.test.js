import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import RegistrationForm from './RegistrationForm';

describe('<App />', () => {
    it('RegistrationForm without crashing', () => {
        shallow(<RegistrationForm />);
    });

    it('renders correctly', () => {
        const tree = renderer
            .create(<RegistrationForm />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});