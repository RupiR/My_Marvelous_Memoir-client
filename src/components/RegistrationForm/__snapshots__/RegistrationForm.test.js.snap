// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly 1`] = `
<form
  className="RegistrationForm"
  onSubmit={[Function]}
>
  <div
    role="alert"
  />
  <div
    className="full_name"
  >
    <label
      htmlFor="RegistrationForm__full_name"
    >
      Full name 
      <span
        className="Required "
      >
        *
      </span>
    </label>
    <input
      className="Input "
      id="RegistrationForm__full_name"
      name="full_name"
      required={true}
      type="text"
    />
  </div>
  <div
    className="username"
  >
    <label
      htmlFor="RegistrationForm__username"
    >
      Username 
      <span
        className="Required "
      >
        *
      </span>
    </label>
    <input
      className="Input "
      id="RegistrationForm__username"
      name="username"
      required={true}
      type="text"
    />
  </div>
  <div
    className="password"
  >
    <label
      htmlFor="RegistrationForm__password"
    >
      Password 
      <span
        className="Required "
      >
        *
      </span>
    </label>
    <input
      className="Input "
      id="RegistrationForm__password"
      name="password"
      required={true}
      type="password"
    />
  </div>
  <div
    className="email"
  >
    <label
      htmlFor="RegistrationForm__email"
    >
      Email 
      <span
        className="Required "
      >
        *
      </span>
    </label>
    <input
      className="Input "
      id="RegistrationForm__email"
      name="email"
      required={true}
      type="text"
    />
  </div>
  <button
    className="Button "
    type="submit"
  >
    Register
  </button>
</form>
`;
